### PR TITLE
LightProbeGenerator: adapt for WebGPU compatibility

### PR DESCRIPTION
**Pull request by aardgoose**
Related issue: #29295

The cubeRenderTarget method has been made async which requires the original example to be modified to account for this.

The cube faces from WebGPU have the 'X' faces swapped, and all mirrored vertically.

As an aside: the WebGPU copyTexturetoBuffer() command can return all 6 faces of a cube texture in a single call in a single buffer.

### CODE REVIEW

File: examples/jsm/lights/LightProbeGenerator.js [hunk 1]
```diff
@@ -150,19 +162,28 @@ class LightProbeGenerator {
...
-			renderer.readRenderTargetPixels( cubeRenderTarget, 0, 0, imageWidth, imageWidth, data, faceIndex );
+				} else {
+
+					// assuming UnsignedByteType
+
+					data = new Uint8Array( imageWidth * imageWidth * 4 );
+
+				}
+
+				renderer.readRenderTargetPixels( cubeRenderTarget, 0, 0, imageWidth, imageWidth, data, faceIndex );
```
Comment (Mugen87) [inline examples/jsm/lights/LightProbeGenerator.js:184]: How about we make this call async as well?

Comment (aardgoose) [reply examples/jsm/lights/LightProbeGenerator.js:184]: I'd forgotten there was an async version for WebGLRenderer. However, I have just tested it, and it seems rather slow, something that isn't replicated using the WebGL fallback oddly.

And while it is not async is there any reason for repeatedly creating the data buffer for each face?

File: examples/jsm/lights/LightProbeGenerator.js [hunk 2]
```diff
@@ -126,7 +127,18 @@ class LightProbeGenerator {
...
+
+		let useAsync = false;
+		let flip = 1;
+
+		if ( renderer.isWebGPURenderer ) {
+
+			useAsync = true;
+
+			if ( renderer.coordinateSystem === WebGPUCoordinateSystem ) flip = -1;
+
```
Comment (WestLangley) [inline examples/jsm/lights/LightProbeGenerator.js:140]: Nit. Perhaps the algorithm can be modified so `flip` can be 1 for WebGPU backend. That way we won't have to be flipping for eternity when WebGL is no longer supported. In other words, let WebGL be the odd case.

File: examples/jsm/lights/LightProbeGenerator.js (hunk 1 above)
Comment (aardgoose) [reply examples/jsm/lights/LightProbeGenerator.js:184]: Also found out that the slowness with WebGLRenderer can be fixed by moving the gl.flush() to after the gl.fenceSync().

File: examples/jsm/lights/LightProbeGenerator.js (hunk 2 above)
Comment (aardgoose) [reply examples/jsm/lights/LightProbeGenerator.js:140]: will do tomorrow

File: examples/jsm/lights/LightProbeGenerator.js (hunk 1 above)
Comment (Mugen87) [reply examples/jsm/lights/LightProbeGenerator.js]: > And while it is not async is there any reason for repeatedly creating the data buffer for each face?

That does indeed seems to be not necessary. Moving this bit out of the for loop is the better approach.

File: examples/jsm/lights/LightProbeGenerator.js (hunk 2 above)
Comment (Mugen87) [reply examples/jsm/lights/LightProbeGenerator.js:140]: I think what @WestLangley suggested is to use `1` as the default value for `flip` and then check for WebGL and set the value to `-1`.

**Comment by Mugen87** [comment]
Do you mind adding an exception for the `webgpu_lightprobe_cubecamera` screenshot here?

https://github.com/mrdoob/three.js/blob/dev/test/e2e/puppeteer.js#L43

I'll revisit this after the merge.

**Pull request merged**